src/promises/allNoLeaks.ts

```diff
@@ -0,0 +1,38 @@
+// From https://stackoverflow.com/questions/64928212/how-to-use-promise-allsettled-with-typescript#69451755
+const isRejected = (
+	input: PromiseSettledResult<unknown>,
+): input is PromiseRejectedResult => input.status === 'rejected';
+
+// From https://stackoverflow.com/questions/64928212/how-to-use-promise-allsettled-with-typescript#69451755
+const isFulfilled = <T>(
+	input: PromiseSettledResult<T>,
+): input is PromiseFulfilledResult<T> => input.status === 'fulfilled';
+
+// The contract of PromiseRejectedResult requires `reason` to be `any`. Thus lint is disabled here.
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const getErrorFromReason = (reason: any): Error => {
+	if (reason instanceof Error) {
+		return reason;
+	}
+	let explanation = 'An unexpected error occurred while awaiting all tasks.';
+	if (typeof reason === 'string') {
```

Comment:
This would only happen if we (or a library we use) made a mistake and threw a string instead of an error, right?

Let's spare ourselves the drama and just not support exposing the content if someone does that particular non-standard behavior.

Reply:
I almost had the simpler form but that would unfortunately swallow whatever error data we may actually have, which in the case that this actually surfaces would be incredibly useful and maddening if not present. An error can occur at any level (our code, our libraries, node, DB, OS, network, hardware) for many reasons. I think what I really want here is the (broad) to-string of `reason` rather than (narrow) case of it being a string.